### PR TITLE
Reverse Engineering Update

### PR DESCRIPTION
**Pull request by Squishy77**
## About the PR
Made the re machine craft able and researchable (alternative research) Added a ton of new rebase items into the re-able list, alongside some side grades like the space heater being now craft able- if wanted. 
## Why / Balance
The re machine was out of date in both recipes and upgrades- this proposes a change in that, now Mutiple things can be red at once and you can logically set up a farm of low difficulty items to research for tech disks to sell to cargo - alongside the side tiers(stuff like space heaters and hand held crew monitors{want to add med-huds and binary keys in the future as a re-able only crafts}) and High tier researches(Blukies and lucky finds of high tier items from salvage or crew). My goal is to make it a reasonable alternative, that can be used to craft items not usually craft able, but not too abuseable in the future, also to give the thief antag another way to obtain tech disks other than the disk printer. Also, I commented out the handheld- teleporter re-ability due to lag it caused (in the past when it was impmented in the research tree) and it not being linked up to a lathe in the code. 
## Technical details
I mostly just extended the re code to the new post rebase items and made the re-machine realistically craftable by changing the diamonds it needed into plasma glass and removing borg arms and the extra glass it needed.

- [x] I have added screenshots/videos to this PR showcasing its changes in game.

image
image
image
image
image
image
image

**Changelog**

- add: made the R.E-machine reasonably craft able, researchable and added a ton of re-able items to it

### CODE REVIEW

File: Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml [hunk 1]
```diff
@@ -28,6 +28,11 @@
     - type: Tag
       tags:
       - WhitelistChameleon
+    - type: ReverseEngineering # Nyano
+      difficulty: 2
+      recipes:
+        - ClothingShoesBootsMagSci
+
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml:35]: one line too many

File: Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml [hunk 2]
```diff
@@ -28,6 +28,11 @@
     - type: Tag
       tags:
       - WhitelistChameleon
+    - type: ReverseEngineering # Nyano
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml]: I'm not *really* sure if this should be tagged Nyano or Delta. Regardless, this should be tagged with a bit more context than just "Nyano" so it's *very* clear to whoever comes next that ReverseEngineering is specifically a Nyano system.

File: Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml [hunk 3]
```diff
@@ -66,6 +71,11 @@
     sprite: Clothing/Shoes/Boots/magboots-science.rsi
   - type: Magboots
     toggleAction: ActionToggleMagbootsSci
+  - type: ReverseEngineering # Nyano
+    difficulty: 2
+    recipes:
+      - ClothingShoesBootsMagSci
+
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml]: one line too many

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 4]
```diff
@@ -66,6 +76,11 @@
         Amount: 1
         DefaultPrototype: Igniter
         ExamineName: Igniter
+  - type: ReverseEngineering # Nyano
+    difficulty: 2
+    recipes:
+      - ProtolatheHyperConvectionMachineCircuitboard
+
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:83]: one line too many

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 5]
```diff
@@ -615,7 +656,6 @@
           DefaultPrototype: Beaker
           ExamineName: Glass Beaker
     - type: ReverseEngineering # Nyano
-      difficulty: 2
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:619]: Why is a heater harder to research than a cloning pod?

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 6]
```diff
@@ -640,7 +680,6 @@
...
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:644]: why is a heater harder to research than a chem dispenser?

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 7]
```diff
@@ -900,7 +948,7 @@
           DefaultPrototype: SaxophoneInstrument
           ExamineName: Woodwind Instrument
     - type: ReverseEngineering # Nyano
-      difficulty: 3
+      difficulty: 2
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:950]: why is a bio fabricator harder to research than this?

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 8]
```diff
@@ -940,6 +988,10 @@
       Capacitor: 4
     materialRequirements:
       Steel: 5
+  - type: ReverseEngineering # Nyano
+    difficulty: 3
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:991]: why is a bio fabricator the same intensity to research as this?

File: Resources/Prototypes/Entities/Objects/Devices/hand_teleporter.yml [hunk 9]
```diff
@@ -9,10 +9,10 @@
     layers:
     - state: icon
   - type: HandTeleporter
-  - type: ReverseEngineering # Nyano
-    difficulty: 4
-    recipes:
-      - HandTeleporter
+  #- type: ReverseEngineering # Nyano
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Devices/hand_teleporter.yml]: This should just be removed or if you want to comment it out then should be tagged with a Delta comment

File: Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml [hunk 10]
```diff
@@ -68,9 +68,10 @@
   - type: StaticPrice
     price: 80
   - type: ReverseEngineering # Nyano
+    difficulty: 3
     recipes:
       - HolofanProjector
-
+
```
Comment (DangerRevolution) [inline Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml]: :O

File: Resources/Prototypes/Research/experimental.yml [hunk 11]
```diff
@@ -56,7 +56,7 @@
   cost: 5000
   recipeUnlocks:
   - TechDiskComputerCircuitboard
-
+  - ReverseEngineeringMachineCircuitboard
```
Comment (DangerRevolution) [inline Resources/Prototypes/Research/experimental.yml]: This needs to be tagged with a Delta comment

**Review by DangerRevolution — COMMENTED** [review]
mostly fine, i really think you should grab all the reverse-engineerable entities into a .txt file and retro-actively go through them to rebalance research difficulty.

File: Resources/Prototypes/Research/experimental.yml (hunk 11 above)
Comment (Squishy77) [reply Resources/Prototypes/Research/experimental.yml]: got it

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml (hunk 5 above)
Comment (Squishy77) [reply Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:619]: that was an acident

File: Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml (hunk 2 above)
Comment (Squishy77) [reply Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml]: yeah makes sense. ill update em

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml (hunk 8 above)
Comment (Squishy77) [reply Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:991]: It's got an intesty of three because it's where it Is in the tech tree

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml (hunk 4 above)
Comment (Squishy77) [reply Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:83]: Will fix

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml (hunk 7 above)
Comment (Squishy77) [reply Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:950]: This is lower on thr reserch teirs, bio fabs are teir two on civ reserch. While daws are teir one

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml (hunk 6 above)
Comment (Squishy77) [reply Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:644]: This is a old change... I made it so they are equal level, as chem dispensers are available reserch , same with circuit printers at round start.

File: Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml (hunk 3 above)
Comment (Squishy77) [reply Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml]: Will fix

File: Resources/Prototypes/Entities/Objects/Devices/hand_teleporter.yml (hunk 9 above)
Comment (Squishy77) [reply Resources/Prototypes/Entities/Objects/Devices/hand_teleporter.yml]: Done!

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml (hunk 5 above)
Comment (Squishy77) [reply Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:619]: I think I fixed it in a latter commit, but it's back at four now... the cloner that is.

**Comment by deltanedas** [comment]
almost all re'd recipes arent available in lathes right now there should probably be a test to prevent this in the future

**Comment by Squishy77** [comment]
What do you mean? I mostly added the ones I knew  that had recipes. You talking about things like the hand held and bluespace parts? I could try to figure them out in the future.

**Comment by deltanedas** [comment]
yes there should be a test that checks if there is a RE'able thing its recipe is in a lathe

**Comment by Squishy77** [comment]
Ah I see, it would be nice .

**Comment by Squishy77** [comment]
Damnit something went wrong with the computer code, I'll dubble check and fix it.

**Comment by Squishy77** [comment]
alright I think it's good now, but again sorry for not noticing that.

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Devices/CircuitBoards/production.yml [hunk 12]
```diff
@@ -33,16 +33,10 @@
...
         Manipulator: 1
       materialRequirements:
-        Glass: 1
         Cable: 1
-        Diamond: 10
-      tagRequirements:
-        BorgArm:
-          Amount: 3
-          DefaultPrototype: LeftArmBorg
-          ExamineName: borg arm
```
Comment (Adrian16199) [inline Resources/Prototypes/Nyanotrasen/Entities/Objects/Devices/CircuitBoards/production.yml:43]: So first things first, why did you delete all the requirements to build the RE besides leaving a singular cable as a requirement.

File: Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml [hunk 13]
```diff
@@ -66,6 +70,11 @@
     sprite: Clothing/Shoes/Boots/magboots-science.rsi
   - type: Magboots
     toggleAction: ActionToggleMagbootsSci
+  - type: ReverseEngineering # delta
+    difficulty: 2
+    recipes:
+      - ClothingShoesBootsMagSci
+
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml]:
```suggestion
```

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 14]
```diff
@@ -662,7 +700,7 @@
           ExamineName: Knife
       materialRequirements:
         Steel: 5
-    - type: ReverseEngineering # Nyano
+    - type: ReverseEngineering # Nyano
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml]:
```suggestion
    - type: ReverseEngineering # Nyano
```

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/computer.yml [hunk 15]
```diff
@@ -227,7 +242,7 @@
   components:
     - type: ComputerBoard
       prototype: ComputerCrewMonitoring
-    - type: ReverseEngineering # Nyano
+    - type: ReverseEngineering # delta
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/computer.yml]: Okay but why
```suggestion
    - type: ReverseEngineering # Nyano
```

File: Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml [hunk 16]
```diff
@@ -67,10 +67,11 @@
       - HolofanProjector
   - type: StaticPrice
     price: 80
-  - type: ReverseEngineering # Nyano
+  - type: ReverseEngineering # delta
+    difficulty: 3
     recipes:
       - HolofanProjector
-
+
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml]:
```suggestion
 
```
I despise that I cant offer to delete 2 spaces instead of one

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 17]
```diff
@@ -972,7 +1027,7 @@
         Glass: 200
       chemicalComposition:
         Silicon: 20
-    - type: ReverseEngineering # Nyano
+    - type: ReverseEngineering # delta
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:1030]:
```suggestion
    - type: ReverseEngineering # Nyano
```
Okay but why: The sequel

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 18]
```diff
@@ -503,9 +539,10 @@
     materialRequirements:
       Cable: 5
       Glass: 2
-  - type: ReverseEngineering # Nyano
+  - type: ReverseEngineering # Nayno
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:543]: This one I will find acceptable.

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 19]
```diff
@@ -777,6 +815,11 @@
         Plastic: 30
     - type: StaticPrice
       price: 30
+    - type: ReverseEngineering # delta
+      difficulty: 3
+      recipes:
+        - PowerCageRechargerCircuitboard
+
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:823]:
```suggestion
```
I forgot to complain about this one.

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/computer.yml [hunk 20]
```diff
@@ -176,6 +188,9 @@
       prototype: ComputerTelevision
     - type: Tag
       tags:
+    - type: ReverseEngineering # delta
+      difficulty: 2
+      recipes:
         - ComputerTelevisionCircuitboard
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/computer.yml:195]: You putted the reverse engineering between the tag which is why the PR's testing doesnt work because round doesnt want to start due to it

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 21]
```diff
@@ -1211,6 +1297,11 @@
       MatterBin: 1
     materialRequirements:
       Steel: 1
+  - type: ReverseEngineering # delta
+    difficulty: 2
+    recipes:
+      - FlatpackerMachineCircuitboard
+
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:1303]:
```suggestion
```
This one aswell

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 22]
```diff
@@ -1051,6 +1110,11 @@
     materialRequirements:
       Glass: 2
       Cable: 5
+  - type: ReverseEngineering # Nyano
+    difficulty: 2
+    recipes:
+      - ElectricGrillMachineCircuitboard
+
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:1117]:
```suggestion
```
This one aswell

File: Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml [hunk 23]
```diff
@@ -67,10 +67,11 @@
...
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml]: this one space is going to be the end of me

File: Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml [hunk 24]
```diff
@@ -67,10 +67,11 @@
       - HolofanProjector
   - type: StaticPrice
     price: 80
-  - type: ReverseEngineering # Nyano
+  - type: ReverseEngineering # delta
```
Comment (Adrian16199) [inline Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml]: Why delta if it was Nyano change here, buh

File: Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml [hunk 25]
```diff
@@ -66,6 +70,10 @@
     sprite: Clothing/Shoes/Boots/magboots-science.rsi
   - type: Magboots
     toggleAction: ActionToggleMagbootsSci
+  - type: ReverseEngineering # delta
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml]: its inherited from parent so can remove

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 26]
```diff
@@ -11,6 +11,9 @@
         Manipulator: 1
       materialRequirements:
         Glass: 1
+    - type: ReverseEngineering # delta
+      recipes:
+        - AutolatheHyperConvectionMachineCircuitboard
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml]: this gives hyperlathe but is lower difficulty than reing an actual hyperlathe

File: Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml [hunk 27]
```diff
@@ -367,6 +392,10 @@
       materialRequirements:
         PlasmaGlass: 5
         Cable: 5
+    - type: ReverseEngineering # delta
+      difficulty: 3
+      recipes:
+       - AnomalyVesselExperimentalCircuitboard
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml]: meant to be synchronizer

File: Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml [hunk 28]
```diff
@@ -126,6 +131,10 @@
         - HolofanProjector
     - type: StaticPrice
       price: 50
+    - type: ReverseEngineering # delta
+      difficulty: 2
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Devices/holoprojectors.yml:135]: is sec projector not roundstart gear

**Review by deltanedas — CHANGES_REQUESTED** [review]
valid

**Review by deltanedas — DISMISSED** [review]

**Review by deltanedas — APPROVED** [review]
merging if doesnt break tests

**Comment by Squishy77** [comment]
damnit what the heck is a saw mill
